src/App.jsx

```diff
@@ -14,45 +16,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_PARAMETER_NAME = 'name';
+const SORT_PARAMETER_LENGTH = 'length';
+
+function getPreparedGoods(goods, sortBy, isReversed) {
+  const preparedGoods = [...goods];
+
+  if (sortBy) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SORT_PARAMETER_NAME:
+          return good1.localeCompare(good2);
+
+        case SORT_PARAMETER_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortBy, setSortBy] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortBy, isReversed);
+  const resetSorting = () => {
+    setSortBy('');
+    setIsReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={
+            cn(
+              'button',
+              'is-info',
+              { 'is-light': sortBy !== SORT_PARAMETER_NAME },
+            )
+          }
+          onClick={() => setSortBy(SORT_PARAMETER_NAME)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={
+            cn(
+              'button',
+              'is-success',
+              { 'is-light': sortBy !== SORT_PARAMETER_LENGTH },
+            )
+          }
+          onClick={() => setSortBy(SORT_PARAMETER_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={
+            cn(
+              'button',
+              'is-warning',
+              { 'is-light': !isReversed },
+            )
+          }
+          onClick={() => setIsReversed(!isReversed)}
+        >
+          Reverse
+        </button>
+
+        {(sortBy || isReversed) && (
```

Comment:
Create a variable for `sortBy || isReversed` condition to avoid combining different logical operators